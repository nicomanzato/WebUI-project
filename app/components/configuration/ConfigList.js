'use strict';

import React from 'react';
import {StyleSheet, Text, View, Button, CheckBox} from 'react-native';
import PropTypes from 'prop-types';
import {StackNavigator} from 'react-navigation';
import ConfigItem from './ConfigItem';

const ConfigList = (props) => (

 <View>
   <View style={styles.titleView}>
     <Text style={styles.titleText}>Silence notifications from ... </Text>
   </View>
   <ConfigItem title="People who have not verified the account" onChange={props.configToggleVerifiedOnly}
               value={props.configVerifiedOnly}/>
   <ConfigItem title="People who do not follow" onChange={props.configToggleDoNotFollow}
               value={props.configDoNotFollow}/>
   <ConfigItem title="People who have default profile information" onChange={props.configToggleHaveDefaultInformation}
               value={props.configHaveDefaultInformation}/>
   <ConfigItem title="Tweets that contain a link" onChange={props.configToggleContainsLink}
               value={props.configContainsLink}/>
   <ConfigItem title="Tweets that have text truncated" onChange={props.configToggleTextTruncated}
               value={props.configTextTruncated}/>
 </View>
);

export default ConfigList;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#FFFF'
  },
  titleView: {
    height: 50,
    backgroundColor: '#627B97',
    marginTop: 1,
    marginBottom: 3,
    paddingLeft: 10,
    justifyContent: 'center',
  },
  titleText: {
    color: 'white',
    fontSize: 16,
    fontWeight: 'bold'
  }
});
