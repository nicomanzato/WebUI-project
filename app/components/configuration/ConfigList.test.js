import React from 'react';
import ConfigList from './configList';
import { shallow, mount, ReactWrapper } from 'enzyme';


describe('config list', () => {

  it('should render correctly', () => {
    const wrapper = shallow(
      <ConfigList
        configVerifiedOnly={false}
        configDoNotFollow={false}
        configHaveDefaultInformation={false}
        configContainsLink={false}
        configTextTruncated={false}

        configToggleVerifiedOnly={() => {}}
        configToggleDoNotFollow={() => {}}
        configToggleHaveDefaultInformation={() => {}}
        configToggleContainsLink={() => {}}
        configToggleTextTruncated={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should dispatch an action when configItem is pressed', () => {

    const mockConfigToggleVerifiedOnly = jest.fn().mockName('mockedFunction');
    const mockConfigToggleDoNotFollow = jest.fn().mockName('mockedFunction');
    const mockConfigToggleHaveDefaultInformation = jest.fn().mockName('mockedFunction');
    const mockConfigToggleContainsLink = jest.fn().mockName('mockedFunction');
    const mockConfigToggleTextTruncated = jest.fn().mockName('mockedFunction');

    const wrapper = mount(
      <ConfigList
        configVerifiedOnly={false}
        configDoNotFollow={false}
        configHaveDefaultInformation={false}
        configContainsLink={false}
        configTextTruncated={false}

        configToggleVerifiedOnly={mockConfigToggleVerifiedOnly}
        configToggleDoNotFollow={mockConfigToggleDoNotFollow}
        configToggleHaveDefaultInformation={mockConfigToggleHaveDefaultInformation}
        configToggleContainsLink={mockConfigToggleContainsLink}
        configToggleTextTruncated={mockConfigToggleTextTruncated}
      />
    );

    const configItems = wrapper.find('ConfigItem');

    configItems.forEach((configItem) => {
      configItem.props().onChange();
    });

    expect(mockConfigToggleVerifiedOnly).toHaveBeenCalled();
    expect(mockConfigToggleDoNotFollow).toHaveBeenCalled();
    expect(mockConfigToggleHaveDefaultInformation).toHaveBeenCalled();
    expect(mockConfigToggleContainsLink).toHaveBeenCalled();
    expect(mockConfigToggleTextTruncated).toHaveBeenCalled();

  });
});
