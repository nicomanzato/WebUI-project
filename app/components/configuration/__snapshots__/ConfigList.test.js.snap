// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config list should render correctly 1`] = `
<Component>
  <Component
    style={
      Object {
        "backgroundColor": "#627B97",
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 3,
        "marginTop": 1,
        "paddingLeft": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Silence notifications from ... 
    </Text>
  </Component>
  <ConfigItem
    onChange={[Function]}
    title="People who have not verified the account"
    value={false}
  />
  <ConfigItem
    onChange={[Function]}
    title="People who do not follow"
    value={false}
  />
  <ConfigItem
    onChange={[Function]}
    title="People who have default profile information"
    value={false}
  />
  <ConfigItem
    onChange={[Function]}
    title="Tweets that contain a link"
    value={false}
  />
  <ConfigItem
    onChange={[Function]}
    title="Tweets that have text truncated"
    value={false}
  />
</Component>
`;
